src/dialects/postgres/connection-manager.js

```diff
@@ -200,6 +200,13 @@ class ConnectionManager extends AbstractConnectionManager {
       });
     });
 
+    // Don't let a Postgres restart (or error) to take down the whole app
```

Comment:
I also considered adding moving this even _further_ up in the lifecycle (prior to `.connect()`), but I was uncertain about whether or not it could potentially interfere with our _other_ error-handling code above.

(Additionally, I don't _think_ there's actually any room for an error to slip in between the establishment of the connection and time that this handler is attached)